### PR TITLE
Develop: Builder methods moved into Rakefile.

### PR DESCRIPTION
**Pull request by philcrissman**
Well; moved most everything needed from make/darwin/tasks.rb into the Rakefile itself. Most of it is namespaced in :osx; other platforms also started with namespacing, but their tasks still point to their respective Builder classes.

There are a few methods pulled straight out of make/make.rb, or make/darwin/tasks.rb; maybe they should become rake tasks?

Removed reference to the darwin/tasks.rb file (see Rakefile:70), everything still compiles on Lion. When I tried removing (or renaming) the actual make/darwin/tasks.rb file, the build failed: so something is evidently referencing the file, but I haven't found what it is.

Feel free to suggest changes, I'm more than happy to move things around and/or clean them up and submit a new pull request later.

### CODE REVIEW

**Comment by wasnotrice** [comment]
Awesome! This builds for me, too. It feels much better.

I have a couple of suggestions:
- let's figure out how to remove make/darwin/tasks.rb before pulling this in, since there's duplicated code there
- I would reword the task names something like this. Maybe some of the tasks in the `build` namespace actually depend on one another. Not sure about that.

``` ruby
  desc "Build for OS X"
  task :build => ["build:skel", "dist/#{NAME}", "dist/VERSION.txt", "build:common", "build:copy_deps", "build:copy_files", "build:setup_system_resources"]

  namespace :build do

    task :skel do
    end

    task :common do
    end

    # and so on

  end
```
- the `osx:common_build` task probably belongs at `common:build` or something like that

**Comment by philcrissman** [comment]
Yeah, I was thinking the same thing; as soon as I did the pull request, I
thought that maybe some more time/thought should have been put into the
organization of the namespaced tasks.

I'll go ahead and do some of that over this weekend and/or early next week
and see what I come up with.
Phil

**Comment by philcrissman** [comment]
I think the rakefile_common can be removed too, it's now redundant (only had a couple lines in it, which I pulled up into the Rakefile directly).

**Comment by steveklabnik** [comment]
I merged in what we have so far. It's still a great improvement, even if it's not 100% there yet!
